editor/editor_vcs_interface.h

```diff
@@ -56,6 +56,52 @@ class EditorVCSInterface : public Object {
 		TREE_AREA_STAGED = 1,
 		TREE_AREA_UNSTAGED = 2
 	};
+	class DiffLine {
+	public:
+		int new_line_no;
+		int old_line_no;
+		String content;
+		String status;
+
+		String old_text;
+		String new_text;
+
+		DiffLine(){};
+	};
+
+	class DiffHunk {
+	public:
+		int new_start;
+		int old_start;
+		int new_lines;
+		int old_lines;
+		List<DiffLine> diff_lines;
+
+		DiffHunk(){};
+	};
+
+	class DiffFile {
+	public:
+		String new_file;
+		String old_file;
+		List<DiffHunk> diff_hunks;
+		DiffFile(){};
+	};
+
+	class Commit {
+	public:
+		String author;
+		String msg;
+		String hex_id;
+		int64_t time;
+	};
+
+	class StatusFile {
+	public:
+		TreeArea area;
+		ChangeType change_type;
+		String file_path;
+	};
```

Comment:
All these are more suited to be `struct` instead of classes since they are only being used for storage, which means you can remove the `public:` because structs have members public by default. Also you can remove the constructors too because you are not doing anything fancy in them.

But this looks much more well organized than before! Good work!